src/v/cluster/cloud_metadata/offsets_snapshot.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "cluster/cloud_metadata/error_outcome.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "outcome.h"
+#include "serde/envelope.h"
+#include "utils/fragmented_vector.h"
+
+namespace cluster::cloud_metadata {
+
+struct group_offsets
+  : public serde::
+      envelope<group_offsets, serde::version<0>, serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+
+    struct partition_offset
+      : public serde::envelope<
+          partition_offset,
+          serde::version<0>,
+          serde::compat_version<0>> {
+        using rpc_adl_exempt = std::true_type;
+        partition_offset(model::partition_id p, kafka::offset o)
+          : partition(p)
+          , offset(o) {}
+        explicit partition_offset() = default;
```

Comment:
nit: no need for explicit unless the constructor takes exactly one argument

Reply:
Done